docs/application_logic/protocol.md

```diff
@@ -1,4 +1,46 @@
+
+Between the Rate Setter and the actual gossip of the message, several steps will take place, but -for the sake of clearness- we ignore these steps for now and return to this subject later. Then, assuming that the message was properly created, it will be propagated to the rest of the network. Since we deal with a large number of nodes, the communication graph cannot be [complete](https://en.wikipedia.org/wiki/Complete_graph). Thus, the [network topology](https://en.wikipedia.org/wiki/Network_topology) will be dictated by the **Neighbor Selection** (aka Auto-Peering) module, described in INSERT LINK. 
+
+![title](Protocol_overview_received_message.png)
+
+We turn our attention now to another point of view: the one of the nodes receiving the new messages, represented in the diagram above. After receiving a message, the node will perform several **syntactical verifications**, that will act like a filter to the messages. Additionally, the message has to be **solidified**, meaning that the node must know all the past cone of the message (i.e., the set of all messages directly or indirectly referenced by the message in question). After this step, the node places all the messages left into an inbox (OUTBOX OR INBOX??). At a fixed global rate (meaning that all nodes use the same rate), the node uses a **scheduler** to choose a message from the inbox. This scheduler works as a gatekeeper, effectively regulating the use of the most scarce resources of the nodes. Since the scheduler works at a fixed rate, the network cannot be overwhelmed. As discussed in INSERT LINK, the scheduler is designed to ensure the following properties:
+
+1. **Consistency**: all honest nodes will schedule the same messages
+2. **Fair access**: the nodes' messages will be scheduled at a fair rate according to their Access Mana
+4. **Bounded latency**: network delay of all messages will be bounded (????????????)
```

Comment:
is that true???

Reply:
4,5 -> 3,4

Reply:
For FPC to work we require some sort of strong partial synchronicity. More specifically all messages and communication must reach the majority of nodes within some bound time. 

i am not aware which other modules require such synchronicity assumptions, but its likely there are some others

Reply:
I agree that we need some synchronicity, what I'm really not sure is that the congestion control guarantees a bounded network delay. I also understand that it guarantees a bounded time spent in the scheduler queues, but how does this imply that the network delay (that has some additional aspects) is always bounded? Doesn't this depend on other network properties?

Reply:
yes you are right. Maybe we should not mention the synchronicity assumptions here then

Reply:
I changed it to "the processing time of all messages will be bounded", that makes more sense to me